python/GafferImageTest/RampTest.py

```diff
@@ -0,0 +1,164 @@
+##########################################################################
+#
+#  Copyright (c) 2018, John Haddon. All rights reserved.
+#  Copyright (c) 2018, Image Engine Design Inc. All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are
+#  met:
+#
+#      * Redistributions of source code must retain the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer in the documentation and/or other materials provided with
+#        the distribution.
+#
+#      * Neither the name of John Haddon nor the names of
+#        any other contributors to this software may be used to endorse or
+#        promote products derived from this software without specific prior
+#        written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+#  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import os
+import unittest
+
+import imath
+
+import IECore
+
+import Gaffer
+import GafferTest
+import GafferImage
+import GafferImageTest
+
+class RampTest( GafferImageTest.ImageTestCase ) :
+
+	def testChannelData( self ) :
+
+		ramp = GafferImage.Ramp()
+		for i, channel in enumerate( [ "R", "G", "B", "A" ] ) :
+			channelData = ramp["out"].channelData( channel, imath.V2i( 0 ) )
+			self.assertEqual( len( channelData ), ramp["out"].tileSize() * ramp["out"].tileSize() )
+
+			expectedValue = 0.994
+			s = GafferImage.Sampler( ramp["out"], channel, ramp["out"]["dataWindow"].getValue() )
+			self.assertAlmostEqual( s.sample( 1919, 1079 ), expectedValue, delta = 0.01 )
+
+	def testFormatHash( self ) :
+
+		# Check that the data hash change when the format does.
+		c = GafferImage.Ramp()
+		c["format"].setValue( GafferImage.Format( 2048, 1156, 1. ) )
+		h1 = c["out"].channelData( "R", imath.V2i( 0 ) ).hash()
```

Comment:
Here we're hashing the data that comes out of the plug. I think we should be testing the hash computed by the node instead :

`h1 = c["out"].channelDataHash( "R", imath.V2i( 0 ) )`